lib/Commands.cc

```diff
@@ -441,15 +442,16 @@ SharedBuffer Commands::newProducer(const std::string& topic, uint64_t producerId
 }
 
 SharedBuffer Commands::newAck(uint64_t consumerId, int64_t ledgerId, int64_t entryId, const BitSet& ackSet,
-                              CommandAck_AckType ackType, CommandAck_ValidationError validationError) {
+                              CommandAck_AckType ackType, uint64_t requestId) {
```

Comment:
The `ConsumerImpl. discardCorruptedMessage ` still passes the validationError to this function. And It will be incorrectly identified as the requestId here.
https://github.com/apache/pulsar-client-cpp/blob/14623ca6f5fbd84a4baf32a2c16673011d85f98c/lib/ConsumerImpl.cc#L828-L829

Reply:
Good find. I think we'd better to use two overloads of `Commands::newAck` to avoid the similar misuse.